sdk/core/Azure.Core/src/Response{T}.cs

```diff
@@ -1,23 +1,31 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.ComponentModel;
 
 namespace Azure
 {
     public class Response<T>
     {
         private readonly Response _rawResponse;
+        private readonly Lazy<T> _value;
 
         public Response(Response response, T parsed)
         {
             _rawResponse = response;
-            Value = parsed;
+            _value = new Lazy<T>(() => parsed);
+        }
+
+        public Response(Response response, Exception exception)
```

Comment:
Instead of taking the `Exception` directly, would it be better to take a `Func<T>` that'll throw the `Exception` for us?  Then we could pass that directly into `Lazy<T>` and potentially get a call stack with a little more context, depending on how folks chose to implement this.

I don't feel strongly about this.

Reply:
I like having more context via call stack, good point.  I'll look at how it changes the calling code.

Reply:
I'm not a fan of having Lazy all together, we don't need either thread-safety guarantees it provides or delayed execution. We should also be careful with rethrowing the same exception instance multiple times. I wonder if subclassing Response might be a better choice here.

Reply:
Maybe and this point we should make Response abstract and add factories, `Response.FromResult<>`/`Response.FromException` etc..

Reply:
Per discussion, we'll do the abstract Response + factories approach.